pss/internal/ticker/ticker.go

```diff
@@ -0,0 +1,54 @@
+package ticker
```

Comment:
I am guessing that the reason for including a new dependency here is quick time travel to replace the original sleeps.


In this case it is being used for a rather simple test. Do you really feel it's warranted to invite external code in just for this?

Reply:
I would like to introduce the practice of injecting the clock as a dependency for each component that requires a time reference throughout Swarm. I missed this in Feeds.

This `tilina/clock` package takes care of offering both a mock clock for tests and a passthrough one for runtime, both implementing the same interface, which is identical to `time`, so people know what to expect. The lib is small and well tested. 

Code that uses `time.Now()` directly cannot be tested reliably. A component should use the injected time reference.

As a result, all these tests run predictably and instantaneously. 

If this is approved, I'd like to refactor Feeds to use it as well.

Reply:
I understand the advantage of it. It's fine by me, and I trust you if you vouch for it.

But this might end up being a discussion of course, so when you say:

> well tested

maybe say something about what that means? Well tested by you? Or are there other sources (than "go report card") that confirm.

To be honest I don't know the protocol for adding new deps to Swarm. I am sure for example @janos does.

Reply:
Well tested means I read the tests and are sufficient. I would have written it myself following the same pattern and include it in Swarm if necessary.

The last thing I want is to start a discussion. If this is approach is not accepted I will remove it.

Reply:
This PR has a new dependency added nicely. There is no problem with that. And for the clock dependency itself, I think that it has very nice api and that it is useful. Go lack of mocking time is very annoying. We can argue if another similar package can be used, given the popularity and development activity for this one compared to others. But I think that it is ok.